### DISCUSSION TITLE
Possible dashboard enhancements

### DISCUSSION THREAD

**Discussion by youzer-name** [jasonacox/Powerwall-Dashboard#428] [Show and tell]
While updating some of my customized dashboards to use the tweaks added in the v4.0.1 standard dashboard,json, I was reminded that I had made some changes to the voltage, frequency, and capacity panels. I thought I'd share them here in case @jasonacox or the rest of the community is interested in adding them to the project.

1) On the Powerwall capacity dashboard, I changed the panel to stack the powerwall current capacity and place that axis on the right. This way you can see the total energy available in all of the powerwalls. I set max/min values so the nominal capacity lines stay near the top of the panel using the left axis. 

The nominal capacity lines are a little closer together than in the original. That could be adjusted by playing with the left axis, but I'm happy with it the way it is.

Original panel on top, my version on bottom:

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/57046429/16167b11-c22e-4ae4-a4e5-97164b9133dd)

2) On the frequency and voltage panels, I added flags for grid outages. It displays a red line along the bottom of the graph whenever there is an 'Off Grid' status. "Off Grid" shows up in the tool tip if you hover over the point, but is not displayed on the legend. 

(Ignore the values on the left axis... I'm using the v4.0.1 panel that shows 240v on the left axis, but I had to go back to an older date to find an off grid event, and there was no 240v data there which threw off that axis)

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/57046429/debae1c8-a331-4cb9-9b7b-950ba7f0c5e1)

Finally, it may not be that important, because hopefully data outages are rare, but I think the 'connect null values' should probably be set to 'threshold' on these panels. I had a period with no vitals data between the last firmware update and today, and I think when there is missing data it's better to show the break in the data rather than connect the dots. The bottom panel below is set for 'threshold, 1 hour'.

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/57046429/35802752-ae27-4f1a-920e-109ad9c6e86e)

I'd be happy to submit a PR for one or both of these changes if there's interest.

**Comment by jasonacox** [comment]
These are great @youzer-name  - please do.
